Xamarin.Forms.Platform.iOS/Renderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,422 @@
+			{
+				try
+				{
+					_avPlayerViewController?.Player?.CurrentItem?.RemoveObserver(_observer, "status");
+				}
+				catch { }
+				finally
+				{
+
+					_observer = null;
+				}
+			}
+		}
+
+		void ObserveStatus(NSObservedChange e)
+		{
+			if (e.NewValue != null)
+			{
+				if (_avPlayerViewController.Player.Status == AVPlayerStatus.ReadyToPlay)
+				{
+					Element?.RaiseMediaOpened();
+				}
+			}
+		}
+
+		void PlayedToEnd(NSNotification notification)
+		{
+			if (Element.IsLooping)
+			{
+				_avPlayerViewController.Player.Seek(CMTime.Zero);
+				_avPlayerViewController.Player.Play();
+			}
+			else
+			{
+				SetKeepScreenOn(false);
+
+				try
+				{
+					Device.BeginInvokeOnMainThread(Element.OnMediaEnded);
+				}
+				catch { }
+			}
+		}
+		
+
+		protected override void OnElementPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+		{
+			switch (e.PropertyName)
+			{
+				case nameof(MediaElement.AreTransportControlsEnabled):
+					_avPlayerViewController.ShowsPlaybackControls = Element.AreTransportControlsEnabled;
+					break;
+
+				case nameof(MediaElement.Source):
+					UpdateSource();
+					break;
+
+				case nameof(MediaElement.CurrentState):
+					switch (Element.CurrentState)
+					{
+						case MediaElementState.Playing:
+							var audioSession = AVAudioSession.SharedInstance();
+							NSError err = audioSession.SetCategory(AVAudioSession.CategoryPlayback);
+							audioSession.SetMode(AVAudioSession.ModeMoviePlayback, out err);
+							err = audioSession.SetActive(true);
+
+							_avPlayerViewController.Player.Play();
+							if (Element.KeepScreenOn)
+							{
+								SetKeepScreenOn(true);
+							}
+							break;
+
+						case MediaElementState.Paused:
+							if (Element.KeepScreenOn)
+							{
+								SetKeepScreenOn(false);
+							}
+							_avPlayerViewController.Player.Pause();
+							break;
+
+						case MediaElementState.Stopped:
+							if (Element.KeepScreenOn)
+							{
+								SetKeepScreenOn(false);
+							}
+							//ios has no stop...
+							_avPlayerViewController.Player.Pause();
+							_avPlayerViewController.Player.Seek(CMTime.Zero);
+
+							err = AVAudioSession.SharedInstance().SetActive(false);
+							break;
+					}
+
+					break;
+
+				case nameof(MediaElement.KeepScreenOn):
+					if (!Element.KeepScreenOn)
+					{
+						SetKeepScreenOn(false);
+					}
+					else if(Element.CurrentState == MediaElementState.Playing)
+					{
+						// only toggle this on if property is set while video is already running
+						SetKeepScreenOn(true);
+					}
+					break;
+
+				case nameof(MediaElement.Aspect):
+					_avPlayerViewController.VideoGravity = AspectToGravity(Element.Aspect);
+					break;
+			}
+
+			base.OnElementPropertyChanged(sender, e);
+		}
+
+		void IMediaElementRenderer.Seek(TimeSpan time)
+		{
+			if (_avPlayerViewController.Player.Status == AVPlayerStatus.ReadyToPlay)
+			{
+				if (_avPlayerViewController.Player.CurrentItem != null)
+				{
```

Comment:
I think it would be better to combine these two conditions into one block and invert it.
```C#
if (_avPlayerViewController.Player.Status != AVPlayerStatus.ReadyToPlay || _avPlayerViewController.Player.CurrentItem == null)
    return;
```